### PR TITLE
Update the plugin with YugabyteDB support of alter commands 

### PR DESCRIPTION
**Pull request by Sfurti-yb**
Problem:
- `Alter table alter column type` statement got skipped during migrations because of the plugin.

Changes made:

- Disabled the plugin code skipping the migrations for v2.19 ad above for which these alter table commands are supported
- Removed test cases that had alter table from expected failures for versions 2.19 and up.
- Changed the minimum db version from postgres version to YugabyteDB version

Testing:

- Tested the changes with a sample app that does alter table column type from varchar to text. Passes for 2.20 now
- Ran the django test suite for versions [2.18](https://jenkins.dev.yugabyte.com/job/users/job/ecosystem-integration-debug/129/) and [2.20](https://jenkins.dev.yugabyte.com/job/users/job/ecosystem-integration-debug/130/)

### CODE REVIEW

File: django_yugabytedb/schema.py [hunk 1]
```diff
@@ -8,19 +8,22 @@
...
+                # Increasing the size of varchar is supported
+                if old_type.lower().startswith("varchar(") and new_type.lower().startswith("varchar("):
+                    conv = lambda i: i or 8
+                    o_len = int(old_type[8:conv(old_type.index(")"))])
+                    n_len = int(new_type[8:conv(new_type.index(")"))])
+                    if o_len > n_len:
+                        logger.warning("Warning: YB does not allow changing the column %s type from %s to %s " % (
+                        old_field, old_type, new_type))
+                        return
+                else:
```
Comment (kneeraj) [inline django_yugabytedb/schema.py:25]: We should file a jira ticket for other data types and follow that up.

**Review by kneeraj — APPROVED** [review]
lgtm

File: django_yugabytedb/schema.py [hunk 2]
```diff
@@ -8,19 +8,22 @@
...
-            # Increasing the size of varchar is supported
-            if old_type.lower().startswith("varchar(") and new_type.lower().startswith("varchar("):
-                conv = lambda i: i or 8
-                o_len = int(old_type[8:conv(old_type.index(")"))])
-                n_len = int(new_type[8:conv(new_type.index(")"))])
-                if o_len > n_len:
-                    logger.warning("Warning: YB does not allow changing the column %s type from %s to %s " % (
-                    old_field, old_type, new_type))
+
+        if self.connection.yugabytedb_version <= (2,18):
```
[the remark this answers is not in the document]
Comment (fizaaluthra) [reply django_yugabytedb/schema.py:12]: It looks like int4 to int8 is also not supported on 2.18, but **varbit(x)** to **varbit(y)** is 
```
yugabyte=# CREATE TABLE x (t varbit(8));
CREATE TABLE
yugabyte=# ALTER TABLE x ALTER COLUMN t TYPE varbit(9);
```

Seems like the logic on DB side is also kind of crude for 2.18.

**Review by fizaaluthra — APPROVED** [review]
Thank you!

File: django_yugabytedb/schema.py (hunk 1 above)
Comment (Sfurti-yb) [reply django_yugabytedb/schema.py:25]: Filed the JIRA ticket for other datatypes: https://yugabyte.atlassian.net/browse/DB-13419

File: django_yugabytedb/schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sfurti-yb) [reply django_yugabytedb/schema.py:12]: Filed a ticket for this datatype, will follow up on it: https://yugabyte.atlassian.net/browse/DB-13419

**Pull request merged**